src/Network/Freddy/Result.hs

```diff
@@ -0,0 +1,8 @@
+module Network.Freddy.Result (Result (..)) where
+
+data Result = Success | Error
```

Comment:
Should probably be `ResultType` instead. You're already using this as `resType` anyway.